ggshield/core/scan/file.py

```diff
@@ -57,65 +54,25 @@ def _read_content(self) -> None:
 def get_files_from_paths(
     paths: List[Path],
     exclusion_regexes: Set[Pattern[str]],
-    yes: bool,
-    display_scanned_files: bool,
-    display_binary_files: bool,
     list_files_mode: ListFilesMode = ListFilesMode.GIT_COMMITTED_OR_STAGED,
-) -> List[Scannable]:
+) -> Tuple[List[Scannable], List[Path]]:
     """
-    Create a scan object from files content.
-
-    :param paths: List of file/dir paths from the command
-    :param yes: Skip confirmation option
-    :param display_scanned_files: In some parts of the code (e.g. SCA), we might want
-    to display a processed list instead and set this to False
-    :param display_binary_files: Display all ignored binary files
-    :param ignore_git: Ignore that the folder is a git repository
+    Returns a tuple of a list of scannables for the files, and a list of ignored binary
+    paths
```

Comment:
```suggestion
    Returns a tuple of a list of scannables for the files, and a list of paths for  binary
    files
```
Nit: at this point, the binary files are not ignored

Reply:
They are ignored in that the function did not create `File` instances for them. So `secret scan` is not going to look at them. This might a bit confusing when looking at `sca|iac scan` because there the code only cares about the paths, it ignores the created File instances.

We should actually be able to make iac|sca code directly use `get_filepaths()` which would be simpler. It only needs to gain support for excluding binaries at that point, but I leave that for future refactorings :).

With this in mind I think the current comment is OK, or maybe it should say "Returns a tuple of a list of **File** instances for the **paths**, and a list of ignored binary paths". What do you think?

Reply:
Ok, I think what confuses me is the name of the function, maybe a clearer name would  be `get_scannables_from_paths`  (in this case, the binary files in the output are not scannable, so they can indeed said to be "ignored")

Reply:
I just went ahead and added a few commits so that iac|sca does not use `get_files_from_paths()` anymore, improved doc and naming. Let me if this looks better. 

(Reviewing commit by commit is going to be simpler)